src/lib/math/pcurves/pcurves_impl/pcurves_impl.h

```diff
@@ -1709,23 +1684,18 @@ inline auto map_to_curve_sswu(const typename C::FieldElement& u) -> typename C::
    const auto tv1 = invert_field_element<C>(z2_u4 + z_u2);
    auto x1 = SSWU_C1<C>() * (C::FieldElement::one() + tv1);
    x1.conditional_assign(tv1.is_zero(), SSWU_C2<C>());
-   const auto gx1 = C::AffinePoint::x3_ax_b(x1);
-
    const auto x2 = z_u2 * x1;
-   const auto gx2 = C::AffinePoint::x3_ax_b(x2);
-
-   // Will be zero if gx1 is not a square
-   const auto [gx1_sqrt, gx1_is_square] = gx1.sqrt();
 
-   auto x = x2;
    // By design one of gx1 and gx2 must be a quadratic residue
-   auto y = gx2.sqrt().first;
+   const auto y1 = sqrt_field_element<C>(C::x3_ax_b(x1));
+   const auto y2 = sqrt_field_element<C>(C::x3_ax_b(x2));
```

Comment:
That may be a bit pedantic; but partially I'd also like to know your opinion on that: Isn't that a place where we should qualify the expected type (i.e. a `CT::Option`) instead of using `auto`? Below, we're using `has_value()` and `value_or()` that both smell like potential side-channel oopsies if these variables would ever become `std::optional<>` for whatever reason.

Reply:
I guess my default response would be if someone is making that kind of change to sqrt, the problem isn't in this function, it's in the change to sqrt. And if someone is making a kind of side-channel unaware change like that, I don't know that having to update a couple of call sites is going to slow them down. That said eh it's easy enough to spell out the type name this 2 times.